lib/spack/spack/compiler.py

```diff
@@ -229,6 +236,9 @@ class Compiler:
     # by any compiler
     _all_compiler_rpath_libraries = ["libc", "libc++", "libstdc++"]
 
+    # string matcher for string representation of platforms supported by a compiler
```

Comment:
A couple things:

* Use `#:` here (which turns this into documentation
* I'm trying to think of a way to link this more substantially to the `Platform` object (i.e. to make it clear to users who want to implement this method in `msvc.py` what this string should be, rather than "win32").  Perhaps clarify here that the string should match `Platform.name`. It could be that passing in the Platform object itself and using `isinstance` would make this more obvious, but either way works for me.

Reply:
1. Will do/done.
2. I'll go with the `isinstance` check, since it avoids string matching